### PR TITLE
S2U-10 5.1.1.6 Tests & Quizzes: Ability to preview calculated questions in question pools

### PR DESCRIPTION
**Pull request by jesusmmp**
https://sakaiproject.atlassian.net/browse/S2U-10

### CODE REVIEW

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties [hunk 1]
```diff
@@ -70,6 +73,7 @@ in_ps=En la(s) bater\u00eda(s)
 last_mod=\u00daltima modificaci\u00f3n
 mv_p=Mover bater\u00eda
 mv_q=Mover pregunta
+sol_q=Solucci\u00f3n
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties:76]: Same value as 'solution', and it says 'Solucción'. let's use one

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties [hunk 2]
```diff
@@ -178,6 +182,7 @@ t_removePool=Eliminar bater\u00eda
 t_copyQuestion=Copiar pregunta
 t_moveQuestion=Mover pregunta
 t_exportQuestion=Exportar pregunta
+t_showSolution=Ver soluci\u00f3n
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties:185]: Mostrar soluci\u00f3n.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java [hunk 3]
```diff
@@ -1249,6 +1254,88 @@ public String startMoveQuestion()
         return "movePool";
   }
 
+  public String checkSolution(){
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java:1257]: Spacing

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java [hunk 4]
```diff
@@ -1249,6 +1254,88 @@ public String startMoveQuestion()
...
+
+		itemBean.setItemData(item);
+		itemBean.setPointsForEdit(item.getScore().toString());
+		itemBean.setMaxPoints(item.getScore());
+		itemBean.setFeedbackValue(item.getCorrectItemFeedbackValue());
+		itemBean.setKey(keysString);
+		itemBean.setItemGradingDataArray(datas);
+
+		FinBean fbean = new FinBean();
+		if(texts.toArray().length>i)
```
Comment (mpellicer) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java:1325]: Spacing

File: samigo/samigo-app/src/webapp/jsf/questionpool/checkSolutionCalculatedQuestion.jsp [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    <h:outputText id="answerKeyMC" value="#{question.key}" escape="false"/>
+  </h:panelGroup>
+  <h:panelGroup>
+    <f:verbatim><br /></f:verbatim>
+    <f:verbatim><b></f:verbatim>
+    <h:outputLabel for="feedSC" value="#{commonMessages.feedback}#{deliveryMessages.column} " />
+    <f:verbatim></b></f:verbatim>
+    <h:outputText id="feedSC" value="#{question.itemData.correctItemFeedbackValue}" escape="false" />
+  </h:panelGroup>
+</h:panelGroup>
```
Comment (mpellicer) [inline samigo/samigo-app/src/webapp/jsf/questionpool/checkSolutionCalculatedQuestion.jsp:65]: EOF

File: samigo/samigo-app/src/webapp/jsf/questionpool/solution.jsp [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+				onclick="SPNR.disableControlsAndSpin(this, null);">
+				<f:actionListener
+					type="org.sakaiproject.tool.assessment.ui.listener.questionpool.CancelPoolListener" />
+				<f:attribute name="returnToParentPool" value="false" />
+			</h:commandButton>
+		</div>
+	</h:form>
+</body>
+	</html>
+</f:view>
```
Comment (mpellicer) [inline samigo/samigo-app/src/webapp/jsf/questionpool/solution.jsp:110]: EOF

**Comment by mpellicer** [comment]
I know some of the Codacy complaints come from copy/paste from other JSF templates, but I believe we can address some of them? They are all medium BTW....but it's true that some comments are printed in the markup.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages.properties [hunk 7]
```diff
@@ -70,6 +73,7 @@ in_ps=In Pool(s)
 last_mod=Last Modified
 mv_p=Move Pool
 mv_q=Move Question
+sol_q=Solution
```
[the remark this answers is not in the document]
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages.properties:75]: Solved

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties (hunk 1 above)
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties:75]: Solved

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties [hunk 8]
```diff
@@ -7,6 +7,9 @@ useradmin=Administraci\u00f3n de usuarios
 confirm=Confirmar
 copy=Copiar
 move=Mover
+solution=Soluci\u00f3n
+another_solution=Otra soluci\u00f3n
+solution_calculated_question = Soluci\u00f3n Pregunta calculada
```
[the remark this answers is not in the document]
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties]: Page header. 
Changed to "Solución de la pregunta calculada" (ES)
Changed to "Solution of the calculated question"  (ENG)

image

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages.properties [hunk 9]
```diff
@@ -177,6 +181,7 @@ t_movePool=Move Pool
 t_removePool=RemovePool
 t_copyQuestion=Copy Question
 t_moveQuestion=Move Question
+t_showSolution=Check Solution
```
[the remark this answers is not in the document]
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages.properties]: Solved

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java (hunk 4 above)
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/questionpool/QuestionPoolBean.java]: Solved

File: samigo/samigo-app/src/webapp/jsf/questionpool/solution.jsp (hunk 6 above)
Comment (jesusmmp) [reply samigo/samigo-app/src/webapp/jsf/questionpool/solution.jsp]: Solved

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties (hunk 2 above)
Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties]: Added all messages on _ca and _eu.
Please review both languages. Thanks! :)

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties (hunk 1 above)
Comment (mpellicer) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/bundle/QuestionPoolMessages_es.properties:75]: I still see solucción, I'll change it if you don't mind!

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
Looks great @jesusmmp ! Let me know when do you want this merged.

**Comment by mpellicer** [comment]
Let's test this in QA, we can always make changes if testing fails.

**Pull request merged**